internal/entities/sensor.go

```diff
@@ -11,26 +11,33 @@ const (
 )
 
 type Sensor struct {
-	ID        int32
+	ID        string
 	CreatedAt time.Time
 	UpdatedAt time.Time
 	Status    SensorStatus
 	Data      []*SensorData
+	Latitude  float64
+	Longitude float64
 }
 
 type SensorData struct {
 	ID        int32
+	SensorID  *string
```

Comment:
We do not need the ID here because `Sensor` has an array of SensorData

Reply:
What about the function `InsertSensorData()` ?

Reply:
I think there was an error there, you used the ID of the sensor data as the ID for the sensor:
```go
	params := &sqlc.InsertSensorDataParams{
			SensorID: d.ID,
			Data:     raw,
		}
```

Reply:
Have been discussed.